### PR TITLE
DS-3579 Context mode and cache management for CLI commands

### PR DESCRIPTION
**Pull request by tomdesair**
This PR is a fix for https://jira.duraspace.org/browse/DS-3579

It improves the performance and memory usage of the following DSpace CLI commands:
- checker
- checker-emailer
- filter-media
- generate-sitemaps
- index-authority
- update-handle-prefix
- export
- import
- itemupdate
- metadata-export
- packager
- bitstore-migrate
- cleanup
- curate
- embargo-lifter
- harvest
- rdfizer
- sub-daily
- doi-organiser

This PR builds and depends on https://github.com/DSpace/DSpace/pull/1694

### CODE REVIEW

File: dspace/config/hibernate-ehcache-config.xml [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+         If the path is a Java System Property it is replaced by
+         its value in the running VM.
+
+         The following properties are translated:
+         user.home - User's home directory
+         user.dir - User's current working directory
+         java.io.tmpdir - Default temp file path -->
+
+    <!-- WARNING: If you are running multiple DSpace instances on the same server, make sure to start
+         each DSpace instance with another value for java.io.tmpdir !!! -->
```
Comment (mwoodiupui) [inline dspace/config/hibernate-ehcache-config.xml]: How to do this?  It's a system property, and every Context in the container will have the same value for it.

**Review by mwoodiupui — APPROVED** [review]
+1 by inspection.  Looks well to me.  It's a lot of code to read, and I can't say that I fully understood every line, but it breaks down into a few patterns that are repeated all over.  The ideas seem sound and the application thorough.  One thing I would add (not in the code) is notes for developers on recommended caching practice when working with Context -- good ways of thinking about what you are coding and how it interacts with the caches.

I'm not sure how to test this, other than a basic smoke test.  I think we will have to live with it and tweak things as we gain experience.

**Comment by pnbecker** [comment]
@tomdesair I would love to see this getting into DSpace 6.1. Did you had some time to enhance it?

**Comment by tdonohue** [comment]
@tomdesair : A friendly reminder that we're getting close to shipping 6.1.  For this PR to receive consideration, we'd need to try and move this forward rapidly (even just as-is, assuming the merge conflicts get cleaned up)

**Comment by tomdesair** [comment]
Normally I've merged all changes of the official dspace-6_x branch. We also fixed the last bug we found. This PR can thus be considered final.

**Comment by terrywbrady** [comment]
+1 by inspection.  I have one suggestion.  It seems confusing to call abort() at the end of a successful process.  Could an alternative method like completeReadOnly() be introduced?

**Comment by mwoodiupui** [comment]
Perhaps complete() should just check readonly and do the Right Thing.

**Review by terrywbrady — APPROVED** [review]
+1, tested locally.  Ran oai index on 500,000 items, index-discovery on 150,000 items and performed a metadata import on 1000 items.  Process was successfully

**Comment by terrywbrady** [comment]
@tomdesair , will you create a port of this PR to master?

**Comment by tomdesair** [comment]
@terrywbrady Yes I will, working on it.

**Pull request merged**